cmd/integration/integration.go

```diff
@@ -295,6 +302,9 @@ func podNotFound(c *client.Client, podNamespace string, podID string) wait.Condi
 func podRunning(c *client.Client, podNamespace string, podID string) wait.ConditionFunc {
 	return func() (bool, error) {
 		pod, err := c.Pods(podNamespace).Get(podID)
+		if apierrors.IsNotFound(err) {
```

Comment:
If the pod is not found that means it's not there "yet".  Returning an error immediately terminates the Poll.  So before, this method ran once and returned failure, so it wasn't waiting.  IsNotFound means "keep looking" not "finish".


> On Mar 23, 2015, at 4:27 PM, Yu-Ju Hong <notifications@github.com> wrote:
> 
> In cmd/integration/integration.go:
> 
> > @@ -295,6 +302,9 @@ func podNotFound(c *client.Client, podNamespace string, podID string) wait.Condi
> >  func podRunning(c *client.Client, podNamespace string, podID string) wait.ConditionFunc {
> >  	return func() (bool, error) {
> >  		pod, err := c.Pods(podNamespace).Get(podID)
> > +		if apierrors.IsNotFound(err) {
> @smarterclayton, why does this help? Line 308 checks whether err is nil anyway...
> 
> —
> Reply to this email directly or view it on GitHub.
>